hydra_redis/redisgraph_demo1.py

```diff
@@ -0,0 +1,348 @@
+        obj1 = api_doc.collections[obj]["collection"].generate()
+        obj_properties["methods"] = str(get_method(obj1["supportedOperation"]))
+        obj_properties["title"] = str(get_title(obj1["supportedProperty"]))
+        response = ur.urlopen(base_url + collection_node.properties[obj])
+        new_file = json.loads(response.read().decode('utf-8'))
+        obj_properties["members"] = str(new_file["members"])
+        obj_collection_node = Node(
+            label="id", alias=obj, properties=obj_properties)
+        redis_graph.add_node(obj_collection_node)
+        edge = Edge(collection_node, "has_collection", obj_collection_node)
+        redis_graph.add_edge(edge)
+        print("commit endpointcollection")
+        redis_graph.commit()
+        print("every collection obj node ", obj_collection_node)
+        connect_nodes(url_node, "has" + obj, obj_collection_node)
+        collectionobjects(
+            obj_collection_node,
+            new_file["members"],
+            new_url,
+            url_node)
+
+
+def get_apistructure(entrypoint_obj, url_node):
+    collection_endpoint = {}
+    classes_endpoint = {}
+    collection = 0
+    classes = 0
+    print("split entrypoint into 2 types of endpoints collection and classes")
+    for obj in url_node.properties:
+        if obj != "@id" and obj != "@type" and obj != "@context":
+            if obj in api_doc.collections.keys():
+                collection = 1
+#                print ("collection ddddd",new_obj)
+                collection_endpoint[obj] = str(url_node.properties[obj])
+            else:
+                classes = 1
+                classes_endpoint[obj] = str(url_node.properties[obj])
+    if collection == 1:
+        collection_node = Node(
+            label="id",
+            alias="collection_endpoint",
+            properties=collection_endpoint)
+        redis_graph.add_node(collection_node)
+        edge = Edge(url_node, "has_collection_endpoint", collection_node)
+        redis_graph.add_edge(edge)
+#        redis_graph.commit()
+        print("collection endpoint node ", collection_node)
+        endpointCollection(collection_node, url_node)
+#        redis_graph.commit()
+    if classes == 1:
+        classes_node = Node(
+            label="id",
+            alias="classes_endpoint",
+            properties=classes_endpoint)
+        redis_graph.add_node(classes_node)
+        edge = Edge(url_node, "has_classes_endpoint", classes_node)
+        redis_graph.add_edge(edge)
+        print("classes endpoint node", classes_node)
+        endpointclasses(classes_node, url_node)
+#        redis_graph.commit()
+
+
+def get_endpoints(entrypoint_obj):
+    print("creating entrypoint node")
+    url_node = Node(label="id", alias="Entrypoint", properties=entrypoint_obj)
+    redis_graph.add_node(url_node)
+    print("commiting")
+    redis_graph.commit()
+    print("entrypoint node ", url_node)
+    get_apistructure(entrypoint_obj, url_node)
+
```

Comment:
You need to update all your code and use `<hydrus.hydraspec.doc_writer.HydraDoc object>` instead of manually parsing the APIDoc.

Reply:
From the graph structure doc you provided, your graph structure looks a bit incorrect. I've suggested some changes. Please take a look https://docs.google.com/document/d/1iQ-0UwRyrePVrmW9z7AuUyK6TiilMIimlhn9msRH3eI/edit

Also, you can start with this basic structure. It'll help you with querying part as well.
![img_20180602_135101](https://user-images.githubusercontent.com/9114274/40872497-c0a6c0b6-666c-11e8-89bc-79f02451181b.jpg)

You need to generate a dynamic graph with a structure similar to this. We can deal with data redundancy later.

Reply:
Moved the graph structure discussion to #18